src/extensions/storage_macro/db_redis.rs

```diff
@@ -0,0 +1,77 @@
+use async_recursion::async_recursion;
+use redis::Commands;
+
+type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+impl crate::application::CoLink {
+    fn _get_con_from_address(&self, redis_address: &str) -> Result<redis::Connection, Error> {
+        let client = redis::Client::open(redis_address)?;
+        let con = client.get_connection()?;
+        Ok(con)
+    }
+
+    async fn _get_con_from_stored_credentials(
+        &self,
+        key_path: &str,
+    ) -> Result<redis::Connection, Error> {
+        let redis_url_key = format!("{}:redis_url", key_path);
+        let redis_url = self.read_entry(redis_url_key.as_str()).await?;
+        let redis_url_string = String::from_utf8(redis_url)?;
+        self._get_con_from_address(redis_url_string.as_str())
+    }
+
+    #[async_recursion]
+    pub(crate) async fn _create_entry_redis(
+        &self,
+        address: &str,
+        key_name: &str,
+        payload: &[u8],
+    ) -> Result<String, Error> {
+        let mut con = self._get_con_from_stored_credentials(address).await?;
+        let response: i32 = con.set_nx(key_name, payload)?;
+        if response == 0 {
+            Err("key already exists.")?
```

Comment:
Can we make sure this is the same type of error as in case the user is expecting the error in the code?